### DISCUSSION TITLE
Haxe repos classified as Nemerle repos

### DISCUSSION THREAD

**Discussion by TheWitheredStriker** [github-linguist/linguist#5986] [Classification] [answered]
Not happening to my repos in particular but I've seen enough misclassified repos for it to be a bit obnoxious, especially when I'm specifically searching for Nemerle projects. 

Essentially, several repositories around the language Haxe are identified as repos based around Nemerle (including when I search for `<search query> language:nemerle`). The languages are nothing alike. Haxe is an ActionScript-based language with cross-compilation features. Nemerle is a .NET language based on C#, but with heavy functional, aspect-oriented and metaprogramming-related features.

Haxe uses the file extensions `.hx`, `.hxproj` and `.hxml`. Nemerle uses `.n`. Several of the Haxe repos I found contain `.n` files that do not contain Nemerle code; they probably aren't even code files, as I see "View raw" when I open them. This is probably what causes the problem.

**Authentic Haxe repos**
- [ds](https://github.com/polygonal/ds)
- [BabylonHx](https://github.com/vujadin/BabylonHx)

**Authentic Nemerle repos**
- [Tao](https://github.com/philiplaureano/Tao)
- [Nor](https://github.com/Ziaw/nor)

**Misclassified Haxe repos**
- [hx-unity3d](https://github.com/AxGord/hx-unity3d)
- [as3tohxfromcaffeine-hx](https://github.com/as3boyan/as3tohxfromcaffeine-hx)

**Comment by lildude** [comment, marked as answer]
🤔 Linguist doesn't classify a repository's language; it classifies the files within it, with the most popular by bytes of code being identified as the primary language of the repo. I suspect you are using this to determine the "repo's language".

Looking at the two misclassified examples, I don't think they're very good examples. Neither have been touched in over 9 years which means they haven't been analysed in over 9 years (we only analyse when a change is made to a repo), so it's quite likely things have changed in terms of how the files are classified.

As it was only two repos, I cloned them and analysed them locally, and sure enough the classification does change:

```console
➜ git clone -q git@github.com:AxGord/hx-unity3d.git
➜ github-linguist hx-unity3d
100.00% 152292     Haxe
➜ git clone -q git@github.com:as3boyan/as3tohxfromcaffeine-hx.git
➜ github-linguist as3tohxfromcaffeine-hx/
57.64%  118550     Haxe
41.99%  86370      ActionScript
0.37%   768        Shell
➜
```

With this in mind, I suspect a lot of the repos you're finding as incorrectly classified haven't been updated in a long time and as such will be showing the classification as determined at the time of the last push.

**Reply by TheWitheredStriker** [reply]
Alrighty, so it's essentially a thing of the past. That's good to hear, thank you.
